### DISCUSSION TITLE
miner start WITHOUT api running

### DISCUSSION THREAD

**Discussion by andiz1014** [filecoin-project/lotus#7825] [Storage Provider]
**Describe the bug**
I have already completed sync for lotus, the state is " complete".
And when I run "lotus-miner run", and check the info of my miner, I got the message of` "ERROR: could not get API info: could not get api endpoint: API not running(no endpoint)"`.
I had already reinit my miner for several times, but it still doesn't run properly.
I think that means the miner start without api running. But the daemon's API runs well.

**To Reproduce**
1. Run 'lotus-miner run'
2. Run 'lotus-miner info'(or 'lotus-miner sectors pledge' , they all not work.)
3. See error"ERROR: could not get API info: could not get api endpoint: API not running(no endpoint)"

**Expected behavior**
Get  the information of my miner without api problem.

**Screenshots**
![image](https://user-images.githubusercontent.com/76585341/103252165-9190e600-49b6-11eb-8716-20f7fc2b1e04.png)
And the firewall of my Linux is closed.
![image](https://user-images.githubusercontent.com/76585341/103252182-a53c4c80-49b6-11eb-8f44-5eb1fce8e5f0.png)

**Version (run `lotus version`):**
Daemon: 1.1.2+git.3908701b7+api0.17.0
Local: lotus version 1.1.2+git.3908701b7

**Additional context**
The internet branch is nerpa.
There was no API problem before. But after I met this problem first time, there was no way to solve it.

**Comment by czw370** [comment]
mine runs the same way.

**Comment by czw370** [comment]
have u fixed it?

**Comment by jburnett** [comment]
One of our miners is experiencing this problem w/v1.5.3.  Miner log is full of message that a sector cannot be terminated b/c no associated PC on chain.  We assume this is preventing API from starting, but can't confirm without a way to clear the sector locally.

```
{"level":"warn","ts":"2021-03-28T15:13:04.019-0400","logger":"sectors","caller":"storage-sealing/fsm.go:509","msg":"sector 19 got error event sealing.SectorTerminateFailed: checking precommit presence: sectorNumber is allocated, but PreCommit info wasn't found on chain"}
```

**Comment by danpintilie** [comment]
Hi! I have the same problem now. Did any of you find a solution?

**Comment by jburnett** [comment]
We continue getting these messages in v1.10.0, but they're fairly benign. The remaining messages are due to local repo with sectors that show [Remove|Terminate]Failed  We have not been able to get the repo cleaned up.  HTH

**Comment by danpintilie** [comment]
And what's to be done? Just wait for a new update? I get this error with v1.10.1 as well.

**Comment by jburnett** [comment]
Our logs are still filled with this warning, and API is running fine.  If your miner API isn't running, it may be due to something else.  Sorry we don't have more help to give.

**Comment by jburnett** [comment]
In case it's helpful, we have tried the following (from memory):

- Terminate sector in TerminateFailed state - returned to TerminateFailed state
- Remove sector in TerminateFailed state - some ended in RemovedState, some went back to TerminateFailed.

**Comment by Reiers** [comment]
Hi @andiz1014 

Thanks for the report. The issue you are experiencing lies elsewhere.

Moving this issue to Lotus Discussions for help and troubleshooting.

Closing ticket.

**Comment by LexLuthr** [comment]
@andiz1014 Do you still need help with this problem in the latest lotus version?

**Reply by LexLuthr** [reply]
I am archiving this discussion as there has been no new input from the author for more than 7 days.
